var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -407,10 +407,16 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     # ###################### Dependencies ##########################
 
     # External Kokkos
-    depends_on("kokkos@4.4.01", when="@master: +kokkos")
-    depends_on("kokkos@4.3.01", when="@16.0.0 +kokkos")
-    depends_on("kokkos@4.2.01", when="@15.1.0:15.1.1 +kokkos")
-    depends_on("kokkos@4.1.00", when="@14.4.0:15.0.0 +kokkos")
+    with when("@14.4: +kokkos"):
+        depends_on("kokkos~complex_align")
+        depends_on("kokkos@4.4.01", when="@master:")
+        depends_on("kokkos@4.3.01", when="@16")
+        depends_on("kokkos@4.2.01", when="@15.1:15")
+        depends_on("kokkos@4.1.00", when="@14.4:15.0")
+        depends_on("kokkos-kernels@4.4.01", when="@master:")
+        depends_on("kokkos-kernels@4.3.01", when="@16")
+        depends_on("kokkos-kernels@4.2.01", when="@15.1:15")
+        depends_on("kokkos-kernels@4.1.00", when="@14.4:15.0")
 
     depends_on("kokkos +wrapper", when="trilinos@14.4.0: +kokkos +wrapper")
     depends_on("kokkos ~wrapper", when="trilinos@14.4.0: +kokkos ~wrapper")
```

Comment:
```suggestion
        depends_on("kokkos+wrapper", when="+wrapper")
        depends_on("kokkos~wrapper", when="~wrapper")
```

can this go before/after the ~complex_align statement?

Reply:
maybe also the cuda_arch / amdgpu_arch bits... sorry for the many requests.

Reply:
Addressed both.